node/pkg/watchers/near/watcher.go

```diff
@@ -1,556 +1,309 @@
-				)
-
-				e.msgChan <- observation
+				e.transactionProcessingQueue.Schedule(newJobs[i], time.Now().Add(newJobs[i].delay))
 			}
 		}
 	}
-
-	return nil
-}
-
-func (e *Watcher) inspectStatus(logger *zap.Logger, hash string, receiver_id string) error {
-	t, err := e.getTxStatus(logger, hash, receiver_id)
-
-	if err != nil {
-		return err
-	}
-
-	return e.parseStatus(logger, t, hash)
 }
 
-func (e *Watcher) lastBlock(logger *zap.Logger, hash string, receiver_id string) ([]byte, uint64, error) {
-	t, err := e.getTxStatus(logger, hash, receiver_id)
-
-	if err != nil {
-		return nil, 0, err
-	}
-
-	last_block := uint64(0)
-
-	outcomes := gjson.ParseBytes(t).Get("result.receipts_outcome")
-
-	if !outcomes.Exists() {
-		return nil, 0, err
-	}
-
-	for _, o := range outcomes.Array() {
-		outcome := o.Get("outcome")
-		if !outcome.Exists() {
-			continue
-		}
+func (e *Watcher) runObsvReqProcessor(ctx context.Context) error {
+	logger := supervisor.Logger(ctx)
 
-		executor_id := outcome.Get("executor_id")
-		if !executor_id.Exists() {
-			continue
-		}
+	supervisor.Signal(ctx, supervisor.SignalHealthy)
 
-		if executor_id.String() == e.wormholeContract {
-			l := outcome.Get("logs")
-			if !l.Exists() {
-				continue
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case r := <-e.obsvReqC:
+			if vaa.ChainID(r.ChainId) != vaa.ChainIDNear {
+				panic("invalid chain ID")
 			}
-			for _, log := range l.Array() {
-				event := log.String()
-				if !strings.HasPrefix(event, "EVENT_JSON:") {
-					continue
-				}
-				logger.Info("event", zap.String("event", event[11:]))
-
-				event_json := gjson.ParseBytes([]byte(event[11:]))
-
-				standard := event_json.Get("standard")
-				if !standard.Exists() || standard.String() != "wormhole" {
-					continue
-				}
-				event_type := event_json.Get("event")
-				if !event_type.Exists() || event_type.String() != "publish" {
-					continue
-				}
 
-				block := event_json.Get("block")
-				if !block.Exists() {
-					continue
-				}
+			txHash := base58.Encode(r.TxHash)
 
-				b := block.Uint()
+			logger.Info("Received obsv request", zap.String("log_msg_type", "obsv_req_received"), zap.String("tx_hash", txHash))
 
-				if b > last_block {
-					last_block = b
-				}
-			}
+			// TODO e.wormholeContract is not the correct value for senderAccountId. Instead, it should be the account id of the transaction sender.
+			// This value is used by NEAR to determine which shard to query. An incorrect value here is not a security risk but could lead to reobservation requests failing.
+			// Guardians currently run nodes for all shards and the API seems to be returning the correct results independent of the set senderAccountId but this could change in the future.
+			// Fixing this would require adding the transaction sender account ID to the observation request.
+			job := newTransactionProcessingJob(txHash, e.wormholeAccount)
+			e.transactionProcessingQueue.Schedule(job, time.Now().Add(-time.Nanosecond))
```

Comment:
The time needs to be the block time...  With what you have here, observation requests will have a different hash/digest then the original message.  In my version, the finalization cache has the block time as well